tests/outputs/PngTest.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace GameOfLife\outputs;
+
+$input = [];
+
+function readline($prompt = null)
+{
+    global $input;
+    if (empty($input))
+    {
+        return "";
+    }
+
+    return array_shift($input);
+}
+
+function expectColor($r, $g, $b)
+{
+    global $input;
+    $input[] = $r;
+    $input[] = $g;
+    $input[] = $b;
+}
```

Comment:
Uoouh, that looks like a [code smell](https://en.wikipedia.org/wiki/Code_smell).
Use global variables always is a little bit fishy. Could you explain for what this method is used for?
Could you rewrite your test-code so that this method is part of the test-class instead?